common/djangoapps/embargo/tests/test_middleware_access_rules.py

```diff
@@ -0,0 +1,444 @@
+"""
+Tests for EmbargoMiddleware with CountryAccessRules
+"""
+
+import mock
+import unittest
+
+from django.db import connection, transaction
+from django.core.urlresolvers import reverse
+from django.conf import settings
+import ddt
+
+from student.tests.factories import UserFactory
+from xmodule.modulestore.tests.factories import CourseFactory
+from xmodule.modulestore.tests.django_utils import (
+    ModuleStoreTestCase, mixed_store_config
+)
+
+# Explicitly import the cache from ConfigurationModel so we can reset it after each test
+from config_models.models import cache
+from embargo.models import (
+    IPFilter, RestrictedCourse, Country, CountryAccessRule, WHITE_LIST, BLACK_LIST
+)
+from django_countries import countries
+
+
+# Since we don't need any XML course fixtures, use a modulestore configuration
+# that disables the XML modulestore.
+MODULESTORE_CONFIG = mixed_store_config(settings.COMMON_TEST_DATA_ROOT, {}, include_xml=False)
+
+
+@ddt.ddt
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class EmbargoCountryAccessRulesTests(ModuleStoreTestCase):
```

Comment:
Great set of tests, but I do wonder what the expected behavior is for white lists, regarding my initial comment. Based on that, there may need to be a new test here.